include/libzfs_impl.h

```diff
@@ -95,6 +95,14 @@ struct zfs_handle {
  */
 #define	ZFS_IS_VOLUME(zhp) ((zhp)->zfs_head_type == ZFS_TYPE_VOLUME)
 
+#ifdef HAVE_SETMNTENT
```

Comment:
We can simplify this further.   Nothing in the current code base ever sets `HAVE_SETMNTENT` so we can distill this down to what's the right interface for FreeBSD and Linux.  It can be re-adapted for illumos in the future as needed.

Reply:
Would you rather this still go via the `MNTTAB_OPEN()`/`MNTTAB_CLOSE()` macros but always defined to fopen/fclose, or just to inline them all?

Reply:
I'd suggest we inline the `fopen() /fclose()` for better readability.  Though, looking at this again I wonder why we can't use `setmntent()` and `endmntent()` here?  This goes back to an early 2010 change did2eadf037f5ae2735bcbc61e3bb2974c6d3235b8e I made, but it's not clear why it was needed at the time.  I believe these functions are available on FreeBSD these days.

Reply:
Both glibc and musl just alias setmntent()/endmntent() to fopen()/fclose() (glibc also appends `ce` mode flags, but we already set `e` and `c` is an internal optimisation).

As for FreeBSD, according to https://github.com/freebsd/freebsd-src at current HEAD (b856b51d149811d68ab9e72daa609f00e13c2ec3):
```
nabijaczleweli@tarta:~/uwu/freebsd-src$ git grep setmntent
contrib/llvm-project/compiler-rt/lib/dfsan/libc_ubuntu1404_abilist.txt:fun:__setmntent=uninstrumented
contrib/llvm-project/compiler-rt/lib/dfsan/libc_ubuntu1404_abilist.txt:fun:setmntent=uninstrumented
sys/contrib/openzfs/cmd/mount_zfs/mount_zfs.c:  fp = setmntent("/etc/mtab", "a+");
sys/contrib/openzfs/lib/libspl/os/linux/getmntany.c:    if ((fp = setmntent(MNTTAB, "r")) == NULL) {
sys/contrib/openzfs/lib/libzfs/libzfs_util.c:   if ((hdl->libzfs_mnttab = setmntent(MNTTAB, "r")) == NULL) {
```

So I think it's best to just go back to fopen()/fclose().